dao/application_authentication_dao.go

```diff
@@ -37,6 +38,21 @@ type applicationAuthenticationDaoImpl struct {
 	UserID   *int64
 }
 
+func (a applicationAuthenticationDaoImpl) getDb(applicationAuthentication *m.ApplicationAuthentication) *gorm.DB {
+	if a.TenantID == nil {
+		panic("nil tenant found in applicationAuthentication db DAO")
+	}
```

Comment:
I like this - but I also think we should throw a `.Where("tenant_id = ?", a.TenantID` on this query before returning it, same with the `UserID` perhaps. 

---

Sidenote: This is where we're going to be adding the `DB.WithContext(a.ctx)` per #455 after that one gets merged. It's shaping up nicely to have all of the "setup" tasks in this method.

Reply:
Also thinking about this some more we could maybe have `getDb()` which just contains the setup functions as well as a `getDbWithTable()` which returns the `getDb().Model(&m.ApplicationAuthention{})`. Can use whichever one needed per method. Curious your thoughts on that?

Reply:
I would personally vote for having a single `getDb()` with:

```go
db := DB.Debug().
    Model(&applicationAuthentication).
    Where("tenant_id = ?", tenantId)

if userResource {
    db.Where("user_id = ?", userId)
}

return db
```

Maybe I missed a query, but I think we're safe in this DAO if we specify the model for all the queries, right?

Reply:
> Also thinking about this some more we could maybe have `getDb()` which just contains the setup functions as well as a `getDbWithTable()` which returns the `getDb().Model(&m.ApplicationAuthention{})`. Can use whichever one needed per method. Curious your thoughts on that?

@lindgrenj6 yes I did it initially now I can't remember why did not used - anyway I added it as you suggested (I just name method with getDbWithModel)

Reply:
> I would personally vote for having a single `getDb()` with:
> 
> ```go
> db := DB.Debug().
>     Model(&applicationAuthentication).
>     Where("tenant_id = ?", tenantId)
> 
> if userResource {
>     db.Where("user_id = ?", userId)
> }
> 
> return db
> ```
> 
> Maybe I missed a query, but I think we're safe in this DAO if we specify the model for all the queries, right?

yes that's right I wanted to split it to more PR - but I did at once -I believe it will be still easy to review.